pkg/cmd/roachtest/decommission.go

```diff
@@ -257,34 +273,92 @@ func execCLI(
 	return string(buf), err
 }
 
-func runDecommissionAcceptance(ctx context.Context, t *test, c *cluster) {
+// runDecommissionRecommission tests a bunch of node
+// decommissioning/recommissioning procedures, all the while checking for
+// replica movement and appropriate membership status detection behavior. We go
+// through partial decommissioning of random nodes, ensuring we're able to undo
+// those operations. We then fully decommission nodes, verifying it's an
+// irreversible operation.
+func runDecommissionRecommission(ctx context.Context, t *test, c *cluster) {
 	args := startArgs("--env=COCKROACH_SCAN_MAX_IDLE_TIME=5ms")
 	c.Put(ctx, cockroach, "./cockroach")
 	c.Start(ctx, t, args)
 
+	// We use a few helpers for random node ID access.
+	var nodeIDs []int
+	for i := 1; i <= c.spec.NodeCount; i++ {
+		nodeIDs = append(nodeIDs, i)
+	}
+	getRandNode := func() int {
+		return nodeIDs[rand.Intn(len(nodeIDs))]
+	}
+	getRandNodeOtherThan := func(ids ...int) int {
+		for {
+			cur := nodeIDs[rand.Intn(len(nodeIDs))]
+			inBlockList := false
+			for _, id := range ids {
+				if cur == id {
+					inBlockList = true
+				}
+			}
+			if inBlockList {
+				continue
+			}
+			return cur
+		}
+	}
+
+	// We define a few short hands to {d,r}ecommission target nodes, through a
+	// given node.
 	decommission := func(
```

Comment:
Could you give this test the "stateful doover"? Add a `type decomTestHelper struct { ... }` and put everything in it that is now referenced by all of these closures, which then become anchored to `*decomTestHelper`.

This allows you to move all of the boilerplate out of the main test, and in particular *below* it (so that readers see the test first and don't have to wade through the setup unless they want to).